core/src/main/java/org/infinispan/persistence/spi/NonBlockingStore.java

```diff
@@ -87,146 +83,145 @@
        * via any methods such as {@link #load(int, Object)}, {@link #publishKeys(IntSet, Predicate)} or
-       * {@link #publishEntries(IntSet, Predicate, boolean)}. It is recommended that a store use the provided
-       * {@link org.infinispan.commons.time.TimeService} in the {@code InitializationContext} to determine if an
-       * entry has expired.
+       * {@link #publishEntries(IntSet, Predicate, boolean)}. Stores should use the provided
+       * {@link org.infinispan.commons.time.TimeService} in the {@code InitializationContext} to determine if entries are
+       * expired.
        * <p>
-       * The information about an entry and its expiration is included in the {@link org.infinispan.metadata.Metadata}
-       * which is accessible from the {@link MarshallableEntry} which is provided.
+       * The information about an entry and its expiration is included in the {@link org.infinispan.metadata.Metadata},
+       * accessible from the {@link MarshallableEntry} that is provided.
        * <p>
        * Stores that have this characteristic must override the {@link #purgeExpired()} method.
        */
       EXPIRATION
    }
 
    /**
-    * The first method that will be invoked to allow the store to be configured and for any additional steps, such as
-    * connecting via a socket or opening file descriptors, to be performed.
+    * The first method to invoke so that the store can be configured and additional steps, such as connecting through
+    * a socket or opening file descriptors, are performed.
     * <p>
     * The provided {@link InitializationContext} contains many helpful objects, including the configuration of the
     * cache and store, concurrency utilities such as {@link org.infinispan.util.concurrent.BlockingManager} or
-    * an executor reserved for non blocking operations only {@link InitializationContext#getNonBlockingExecutor()}.
+    * an executor reserved for non-blocking operations only {@link InitializationContext#getNonBlockingExecutor()}.
     * <p>
-    * This method is guaranteed to not be invoked concurrently with other operations. This means another method
-    * will not be invoked on this store until after the returned Stage completes.
+    * This method is guaranteed not to be invoked concurrently with other operations. This means other methods are
+    * not invoked on this store until after the returned Stage completes.
     * <p>
     * It is expected that an implementation should be able to "restart" by invoking {@code start} a second time if
     * {@link #stop()} has been invoked and allowed for its stage to complete.
-    * @param ctx initialization context used to initialize this store
-    * @return a stage that when complete signals that this store has been successfully started
+    * @param ctx initialization context used to initialize this store.
+    * @return a stage that, when complete, signals that this store has started successfully.
```

Comment:
@wburns here do we lose meaning if we change "signals" to "indicates"? or is "signals" terminology that is consistent with rxjava etc?

Reply:
No, signal is just a term I used here. This is specific to CompletionStage as well at this point, so I am fine either way.